### PR TITLE
Enable universal2 wheels

### PR DESCRIPTION
**Pull request by judahrand**
Changes ported from `numpy-wheels`  found at:
- https://github.com/MacPython/numpy-wheels/pull/104
- https://github.com/MacPython/numpy-wheels/pull/115

### CODE REVIEW

File: config.sh [hunk 1]
```diff
@@ -16,12 +16,29 @@ function build_wheel {
     fi
```
Comment (isuruf) [inline config.sh:16]: In the line above, change `build_osx_wheel $@` to `wrap_wheel_builder build_osx_wheel $@`

**Comment by isuruf** [comment]
You also need https://github.com/MacPython/numpy-wheels/pull/109 and https://github.com/MacPython/numpy-wheels/pull/110

File: config.sh (hunk 1 above)
Comment (judahrand) [reply config.sh:16]: Changed, thank you!

**Comment by judahrand** [comment]
> You also need [MacPython/numpy-wheels#109](https://github.com/MacPython/numpy-wheels/pull/109) and [MacPython/numpy-wheels#110](https://github.com/MacPython/numpy-wheels/pull/110)

Should be included now, thanks very much for pointing those out.

**Comment by judahrand** [comment]
I believe the PR I opened on `scipy` is also needed for the `universal2` builds?

File: config.sh [hunk 2]
```diff
@@ -12,16 +12,38 @@ function build_wheel {
...
+    local py_osx_ver=${2:-$(macpython_sdk_for_version $py_version)}
+    local inst_ext=$(pyinst_ext_for_version $py_version)
+    echo "python-${py_version}-macosx${py_osx_ver}.${inst_ext}"
+}
+
+function install_delocate {
+    check_pip
+    $PIP_CMD install git+https://github.com/isuruf/delocate.git@arm64
+}
+
```
Comment (isuruf) [inline config.sh:39]: If you update multibuild, this can be removed.

**Comment by rgommers** [comment]
Something is still wrong, on TravisCI:
```
Could not download "https://anaconda.org/multibuild-wheels-staging/openblas-libs/v0.3.17/download/openblas-v0.3.17-manylinux1_aarch64.tar.gz"
```
It should be `manylinux2014` instead of `manylinux1`

**Comment by rgommers** [comment]
Appveyor is also unhappy:
```
could not find library 'quadpack' in directories ['C:\\projects\\scipy-wheels\\scipy\\build\\openblas_lapack_info']
could not find library 'mach' in directories ['C:\\projects\\scipy-wheels\\scipy\\build\\openblas_lapack_info']
could not find library 'openblas_lapack_info' in directories ['C:\\projects\\scipy-wheels\\scipy\\build\\openblas_lapack_info']
C:\mingw-w64\x86_64-6.3.0-posix-seh-rt_v5-rev1\mingw64\bin\gfortran.exe -Wall -g -Wall -g -shared -Wl,--whole-archive C:\Python39-x64\lib\openblas.a -Wl,--no-whole-archive -LC:\mingw-w64\x86_64-6.3.0-posix-seh-rt_v5-rev1\mingw64\lib\gcc\x86_64-w64-mingw32\6.3.0 -LC:\Python39-x64\libs -LC:\Python39-x64\PCbuild\amd64 -Lbuild\temp.win-amd64-3.9 -o build\temp.win-amd64-3.9\Release\.libs\libopenblas.XWYDX2IKJW2NMTWSFYNGFUWKQU3LYTCZ.gfortran-win_amd64.dll -Wl,--allow-multiple-definition -Wl,--output-def,build\temp.win-amd64-3.9\Release\libopenblas.XWYDX2IKJW2NMTWSFYNGFUWKQU3LYTCZ.gfortran-win_amd64.def -Wl,--export-all-symbols -Wl,--enable-auto-import -static -mlong-double-64
C:\Python39-x64\lib\openblas.a(cblas_xerbla.obj):xerbla.c:(.text+0xe): undefined reference to `__imp___acrt_iob_func'
...
collect2.exe: error: ld returned 1 exit status
error: Command "C:\mingw-w64\x86_64-6.3.0-posix-seh-rt_v5-rev1\mingw64\bin\gfortran.exe -Wall -g -Wall -g -shared -Wl,--whole-archive C:\Python39-x64\lib\openblas.a -Wl,--no-whole-archive -LC:\mingw-w64\x86_64-6.3.0-posix-seh-rt_v5-rev1\mingw64\lib\gcc\x86_64-w64-mingw32\6.3.0 -LC:\Python39-x64\libs -LC:\Python39-x64\PCbuild\amd64 -Lbuild\temp.win-amd64-3.9 -o build\temp.win-amd64-3.9\Release\.libs\libopenblas.XWYDX2IKJW2NMTWSFYNGFUWKQU3LYTCZ.gfortran-win_amd64.dll -Wl,--allow-multiple-definition -Wl,--output-def,build\temp.win-amd64-3.9\Release\libopenblas.XWYDX2IKJW2NMTWSFYNGFUWKQU3LYTCZ.gfortran-win_amd64.def -Wl,--export-all-symbols -Wl,--enable-auto-import -static -mlong-double-64" failed with exit status 1
```

**Comment by judahrand** [comment]
I'll try to have a look at this this evening

**Comment by rgommers** [comment]
Continued in gh-125, closing this PR.

**Pull request closed without merging**